rewrite-java/src/main/java/org/openrewrite/java/internal/PrintJava.java

```diff
@@ -433,9 +433,11 @@ public String visitNewArray(NewArray newArray) {
 
     @Override
     public String visitNewClass(NewClass newClass) {
+        String encl = newClass.getEncl() == null ? "" : visit(newClass.getEncl()) + ".";
+        String nooh = newClass.getNooh() == null ? "" : newClass.getNooh().getFormatting().getPrefix();
```

Comment:
@jkschneider we think this is right but it doesn't follow the visitor pattern like all the other elements do. Is this fine or is there something else we should be doing?

Reply:
`fmt(newClass.getNooh(), "")` is slightly different and may prevent a future bug where some visitor adds a suffix to the `NewClass.New` (even though the original parser never places such a suffix).